### ISSUE TITLE
Cube example Vulkan validation layer errors

### ISSUE THREAD

**Issue by junglie85** [BVE-Reborn/rend3#438]
When I try to run the `cube` example for commit [1fc492e](https://github.com/BVE-Reborn/rend3/commit/1fc492eedd9deebca80063a3985ff057d57f776a) I get a whole host of Vulkan validation layer errors:

```
Vulkan Instance Version: 1.3.204
```

```
     Running `target\debug\cube.exe`
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 163.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 163.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 163.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 163.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 163.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 163.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 492.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 492.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 492.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 492.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 492.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
        Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-Flat-06202] OpEntryPoint interfaces variable must not be vertex execution model with an input storage class for Entry Point id 492.
      %object_idx = OpVariable %_ptr_Input_uint Input

[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-VkSwapchainCreateInfoKHR-imageExtent-01274 (0x7cd0911d)]
        Validation Error: [ VUID-VkSwapchainCreateInfoKHR-imageExtent-01274 ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x7cd0911d | vkCreateSwapchainKHR() called with imageExtent = (800,600), which is outside the bounds returned by vkGetPhysicalDeviceSurfaceCapabilitiesKHR(): currentExtent = (3840,2054), minImageExtent = (3840,2054), maxImageExtent = (3840,2054). The Vulkan spec states: imageExtent must be between minImageExtent and maxImageExtent, inclusive, where minImageExtent and maxImageExtent are members of the VkSurfaceCapabilitiesKHR structure returned by vkGetPhysicalDeviceSurfaceCapabilitiesKHR for the surface (https://vulkan.lunarg.com/doc/view/1.3.224.1/windows/1.3-extensions/vkspec.html#VUID-VkSwapchainCreateInfoKHR-imageExtent-01274)
[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
[ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-VkSwapchainCreateInfoKHR-imageExtent-01274 (0x7cd0911d)]
        Validation Error: [ VUID-VkSwapchainCreateInfoKHR-imageExtent-01274 ] Object 0: handle = 0x2058d3a5c80, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x7cd0911d | vkCreateSwapchainKHR() called with imageExtent = (800,600), which is outside the bounds returned by vkGetPhysicalDeviceSurfaceCapabilitiesKHR(): currentExtent = (3840,2054), minImageExtent = (3840,2054), maxImageExtent = (3840,2054). The Vulkan spec states: imageExtent must be between minImageExtent and maxImageExtent, inclusive, where minImageExtent and maxImageExtent are members of the VkSurfaceCapabilitiesKHR structure returned by vkGetPhysicalDeviceSurfaceCapabilitiesKHR for the surface (https://vulkan.lunarg.com/doc/view/1.3.224.1/windows/1.3-extensions/vkspec.html#VUID-VkSwapchainCreateInfoKHR-imageExtent-01274)
[ERROR wgpu_hal::vulkan::instance]      objects: (type: DEVICE, hndl: 0x2058d3a5c80, name: ?)
```

**Comment by cwfitzgerald** [comment]
This is a wgpu bug which is fixed in 0.14 (naga 0.10): https://github.com/gfx-rs/naga/issues/2036. Drivers don't care in practice, but when the vertex-pulling branch is done (including the wgpu update) this will be fixed.

**Comment by cwfitzgerald** [comment]
Closed by #449
